optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/acceptor/greatdeluge/GreatDelugeAcceptor.java

```diff
@@ -0,0 +1,118 @@
+package org.optaplanner.core.impl.localsearch.decider.acceptor.greatdeluge;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.localsearch.decider.acceptor.AbstractAcceptor;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchMoveScope;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchStepScope;
+import org.optaplanner.core.impl.score.ScoreUtils;
+
+public class GreatDelugeAcceptor extends AbstractAcceptor {
+
+    private Score initialLevel = null;
+
+    private Double rainSpeed = null;
+    private Double rainSpeedRatio = DEFAULT_RAIN_SPEED_RATIO;
+
+    private int levelsLength = -1;
+    private double[] initialLevelScoreLevels;
+    private double[] levelScoreLevels;
+
+    private double levelMinimum = 0;
```

Comment:
(serious issue) The levelMinimum is the same for all levels - this basically presumes that all constraints are the same sign and are somehow in the same scale. In reality, most use cases have wildly different scales between hard and soft (for example 7 hard broken and 1234567 soft broken - the latter representing money) so that won't give a problem as long as the minimum is 0. As for the sign: some example use positive constraints and have a positive soft score.

Reply:
I didn't know about that. I thought the scoring function value is always negative and tries to maximize (approach 0). So what should be used as a breakpoint? There have to be some. For example, a function with the initial score of -10hard / -1000soft must increase of hard score end on 0, otherwise, none of the next steps can be accepted, because it can find best solutions with 0hard/...soft.
Can you please give me some direction to go?

Reply:
Wouldn't it make sense to do the same thing that SA does? I'm assuming SA has to solve the same problem - correct, @ge0ffrey ?